core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -368,8 +368,9 @@ private boolean manifestHasDeletedFiles(
           deletePaths.contains(file.path())
               || dropPartitions.contains(file.specId(), file.partition())
               || (isDelete
-                  && entry.sequenceNumber() > 0
-                  && entry.sequenceNumber() < minSequenceNumber);
+                  && entry.isLive()
```

Comment:
Attention! I think a delete manifest has files to delete ONLY if the entry status is ADDED or EXISTING. Also, we can't use `entry.dataSequenceNumber()` otherwise as it may return null for old V2 manifests with DELETED entries.

Reply:
This looks good to me.